packages/core/ios/Tests/MockDatadogCore.swift

```diff
@@ -0,0 +1,45 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+@testable import DatadogCore
+@testable import DatadogInternal
+
+internal class MockDatadogCore: DatadogCoreProtocol {
+    func set(baggage: @escaping () -> DatadogInternal.FeatureBaggage?, forKey key: String) {}
+    
+    func send(message: FeatureMessage, else fallback: @escaping () -> Void) {
+        if  // Configuration Telemetry Message
+            case .telemetry(let telemetry) = message,
+            case .configuration(let configuration) = telemetry {
+            self.configuration = configuration
+        }
+        
+        if case .baggage(let key, let baggage) = message {
+            self.baggages[key] = baggage
+        }
+    }
+   
+    private(set) var configuration: ConfigurationTelemetry?
+    private(set) var features: [String: DatadogFeature] = [:]
+    private(set) var baggages: [String: Any] = [:]
```

Comment:
I can see it's not a new class, but all of these can be accessed from different threads, you can use `@ReadWriteLock` annotation to make them thread-safe, such as:
```swift
@ReadWriteLock
private(set) var baggages: [String: Any] = [:]
```